src/test/java/com/commercetools/project/sync/customer/CustomerSyncerTest.java

```diff
@@ -55,4 +71,67 @@ void getQuery_ShouldBuildCustomerQuery() {
     final CustomerQuery query = customerSyncer.getQuery();
     assertThat(query).isEqualTo(CustomerReferenceResolutionUtils.buildCustomerQuery());
   }
+
+  @Test
+  void syncWithError_ShouldCallErrorCallback() {
+    // preparation: customer with no key is synced
+    final SphereClient sourceClient = mock(SphereClient.class);
+    final SphereClient targetClient = mock(SphereClient.class);
+    when(sourceClient.getConfig()).thenReturn(SphereApiConfig.of("source-project"));
+    when(targetClient.getConfig()).thenReturn(SphereApiConfig.of("target-project"));
+    final List<Customer> customers =
+        Collections.singletonList(readObjectFromResource("customer-no-key.json", Customer.class));
+
+    final PagedQueryResult<Customer> pagedQueryResult = mock(PagedQueryResult.class);
+    when(pagedQueryResult.getResults()).thenReturn(customers);
+    when(sourceClient.execute(any(CustomerQuery.class)))
+        .thenReturn(CompletableFuture.completedFuture(pagedQueryResult));
+
+    // test
+    final CustomerSyncer customerSyncer =
+        CustomerSyncer.of(sourceClient, targetClient, mock(Clock.class));
+
+    // assertion
+    customerSyncer.sync(null, true).toCompletableFuture().join();
+    final LoggingEvent errorLog = syncerTestLogger.getAllLoggingEvents().get(0);
+    assertThat(errorLog.getMessage())
```

Comment:
how does it look like the error message coming from an error callback?

Reply:
Do you mean this? https://github.com/commercetools/commercetools-project-sync/pull/193/files/a3322dcf32d0e217e753b5249b6c2cae851dd9de#diff-b80b4644f425dc2d5c9a91c50d59f6a22475b1f5aed3a7e23f96a1ca74cdf8ceR99

Reply:
No, an error message coming with `commercetools-sync-java` library.

Reply:
https://github.com/commercetools/commercetools-project-sync/pull/193/commits/da1b928bdb2dfd9f9256c449a0739dff705b93fe